wavefront/server/packages/flo_cloud/flo_cloud/azure/blob_storage.py

```
@@ -0,0 +1,353 @@
+
+        Raises:
+            CloudStorageFileNotFoundError: If the blob or container does not exist.
+            Exception: For other errors.
+        """
+        try:
+            blob_client = self.client.get_blob_client(
+                container=bucket_name, blob=file_path
+            )
+            return blob_client.download_blob().readall()
+        except ResourceNotFoundError:
+            raise CloudStorageFileNotFoundError(bucket_name, file_path)
```

Ruff (0.15.6)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)